workers/unity/Packages/com.improbable.gdk.core/Config/LocatorConfig.cs

```diff
@@ -55,16 +74,51 @@ public void SetProjectName(string projectName)
         }
 
         /// <summary>
-        ///     Sets the login token for the Locator to use.
+        ///     Sets the LocatorCredentialsType to LoginToken and sets the login token for the Locator to use.
         /// </summary>
+        /// <remarks>
+        ///     SpatialOS login token and Steam credentials may not be set at the same time when connecting to a deployment.
+        /// </remarks>
         /// <param name="loginToken">The login token.</param>
+        /// <exception cref="ConnectionFailedException">
+        ///     Thrown if this method is called after Steam credentials were set before.
+        /// </exception>
         public void SetLoginToken(string loginToken)
         {
+            if (LocatorParameters.CredentialsType == LocatorCredentialsType.Steam)
+            {
+                throw new ConnectionFailedException(
+                    "SpatialOS login token and Steam credentials may not be set at the same time when connecting to a deployment.",
+                    ConnectionErrorReason.InvalidConfig);
+            }
+
             LocatorParameters.CredentialsType = LocatorCredentialsType.LoginToken;
-            LocatorParameters.LoginToken = new LoginTokenCredentials
+            LocatorParameters.LoginToken.Token = loginToken;
```

Comment:
does that always work? what if `LocatorParameters.LoginToken` is null?

Reply:
A `null` token should be fine, i.e. it's worker's responsibility to ensure that this doesn't e.g. crash the worker. A null or empty string token would lead to an unsuccessful connection but so would any other invalid token string.

Reply:
oh no, I mean not if it's set to null, but what if 
`LocatorParameters.LoginToken` is null when you are trying to set `LocatorParameters.LoginToken.Token`

Reply:
this will throw an exception on our side

Reply:
Ah, I get it now. That's going to be fine. The `LogingToken` and `Steam` fields of a `LocatorParameters` are initialised and not null by default. :)

Reply:
Ah, I get it now. That should be fine. The `LoginToken` and `Steam` fields of a `LocatorParameters` object are initialised and not null by default. :)

Reply:
Ah, I get it now. That should be fine. The `LoginToken` and `Steam` fields of a `LocatorParameters` object are initialised and not null by default. :)

Reply:
Ah, I get it now. That should be fine. The `LoginToken` and `Steam` fields of a `LocatorParameters` object are initialised and not null by default. :)

Reply:
Ah, I get it now. That should be fine. The `LoginToken` and `Steam` fields of a `LocatorParameters` object are initialised and not null by default. :)

Reply:
Ah, I get it now. That should be fine. The `LoginToken` and `Steam` fields of a `LocatorParameters` object are initialised and not null by default. :)

Reply:
Ah, I get it now. That should be fine. The `LoginToken` and `Steam` fields of a `LocatorParameters` object are initialised and not null by default. 🙂

Reply:
Ah, I get it now. That should be fine. The `LoginToken` and `Steam` fields of a `LocatorParameters` object are initialised and not null by default. 🙂